Detailed Example of Arch (#24)
cheneyzhoucheng/network-digital-twin: draft-irtf-nmrg-network-digital-twin-arch.md

# A Reference Architecture of Network Digital Twin  {#arch}

Considerable industry work and research has focused on
automation-supporting network systems. As an example,
{{ETSI-GS-ZSM-002}} describes a framework architecture for network
automation. It uses so-called management services as a fundamental
conceptual unit of currency, and describes the enablement of automation
use cases through flexible composition and extension of such management
services. For example, a closed loop might be represented as a composition
of appropriate data, analytics, intelligence/decision and orchestration/control
services.

The role and and utility of NDTs may be represented architecturally by

> following similar principles, e.g. {{ETSI-GS-ZSM-015}}. As described in section 4,

Reviewer: ```suggestion
following similar principles, e.g., {{ETSI-GS-ZSM-015}}. As described in Section 4,
```

> an NDT may be represented as encompassing models, data, mapping and interfaces;

Reviewer: ```suggestion
an NDT may be represented as encompassing models, data, mapping, and interfaces;
```

these components then work together and in composition with appropriate other
functions or services to deliver overall functional architectures matching specific
use cases.

For example, see Figure 2. The NDT (or, NDT core or layer) is represented as
encompassing data and models, along with capability to manage them together to
useful purpose. The overall system architecture shows an intent-capable controller
mediating between intent-generating network service demand sources and the physical
network. The NDT plays an important role in this controller. An “outer” closed loop
detects gaps between service parameters set by intents and actual service characteristics,
finds solutions to close those gaps, and drives those solutions on to the network. Finding

> solutions makes use of an “inner loop” that includes an NDT: effectively, prospective solutions

Reviewer: ```suggestion
solutions makes use of an "inner loop" that includes an NDT: effectively, prospective solutions
```

> are postulated, their impacts on services are evaluated by the NDT acting as a “sandbox” in

Reviewer: ```suggestion
are being proposed, their impacts on services are evaluated by the NDT acting as a "sandbox” in
```

     are exchanged through a northbound interface, so they are applied
     by service emulation at the appropriate twin instance(s).

Real Network: All or subset of network elements in the real network exchange
network data and control messages with a network digital twin instance, through
twin-real control interfaces. The real network can be a mobile access network,
a transport network, a mobile core, a backbone, etc. The real network can also
be a data center network, a campus enterprise network, an industrial Internet of
Things, etc.

> The real network can span across a single network administrative domain or multiple

Reviewer: ```suggestion
: The real network can span across a single network administrative domain or multiple
```

network administrative domains. The real network can include both physical entities

> and some virtual entities (e.g. vSwitches, NFVs, etc.), which together carry traffic

Reviewer: ```suggestion
and some virtual entities (VNFs, for example), which together carry traffic
```

and provide actual network services.

> This document focuses on the IETF related real networks such as IP bearer and data

Reviewer: ```suggestion
: This document focuses on the IETF related real networks such as IP bearer and data
```
